### PR TITLE
Ot275 45: endpoint eliminación de categorías

### PR DESCRIPTION
**Pull request by FedericoRohr**
# [Ot275-45](https://alkemy-labs.atlassian.net/browse/Ot275-45)

-Added delete of categories
-classes added : IDeleteCategoryUseCase , DeleteCategoryUseCase,CategoryResourse.
-added repository methods exists and delete (does softdelete)
-Unit test added

### HOW HAS THIS BEEN TESTED?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration.

- [X ] Postman.
- [X ] Unit test.
- [ ] Integration test.
- [ ] No testing required (only applies for tech task).

**Test Configuration**:

* ...
* ...

**Screenshots**:
image
image
image
image

* ....

### CHECKLIST

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added the new resource in the Postman Collection file.

### CODE REVIEW

File: src/test/java/com/alkemy/ong/application/service/DeleteCategoryUseCaseServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.alkemy.ong.application.repository.ICategoryRepository;
+import com.alkemy.ong.domain.Identifiable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class DeleteCategoryUseCaseServiceTest {
```
Comment (mkain0) [inline src/test/java/com/alkemy/ong/application/service/DeleteCategoryUseCaseServiceTest.java:17]: El modificador de acceso `public` no es necesario en este contexto.

File: src/main/java/com/alkemy/ong/infrastructure/database/repository/abstraction/ICategorySpringRepository.java [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
 
 @Repository
 public interface ICategorySpringRepository extends JpaRepository<CategoryEntity, Long> {
 
+  @Modifying
+  @Query("UPDATE CategoryEntity C SET C.softDeleted = true WHERE C.categoryId = :id")
+  void softDelete(@Param("id") Long id);
+
+  @Query(value = "SELECT C FROM CategoryEntity C "
+      + "WHERE C.softDeleted = false AND C.categoryId = :id")
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/database/repository/abstraction/ICategorySpringRepository.java:20]: c minúscula para ser consistente con los el código de los otros repositorios.

**Review by mkain0 — CHANGES_REQUESTED** [review]
Por favor, resolver los merge conflicts antes de volver a solicitar review.

**Comment by FedericoRohr** [comment]
solucionando los merge conflict encontre que faltaba una "d" en la entity de members. Tuve que agregarla para poder pushear mi  ticket

**Review by mkain0 — APPROVED** [review]

**Pull request merged**